dart/lib/src/hub.dart

```diff
@@ -32,20 +35,18 @@ class Hub {
     return Hub._(options);
   }
 
-  Hub._(SentryOptions options)
-      : _options = options,
-        _stack = ListQueue() {
+  Hub._(SentryOptions options) : _options = options {
     _stack.add(_StackItem(_getClient(fromOptions: options), Scope(_options)));
     _isEnabled = true;
   }
 
   static void _validateOptions(SentryOptions options) {
     if (options == null) {
-      throw ArgumentError.notNull('options');
+      throw ArgumentError.notNull('SentryOptions is required.');
     }
 
-    if (options.dsn == null) {
-      throw ArgumentError.notNull('options.dsn');
+    if (options.dsn == null || options.dsn.isEmpty) {
```

Comment:
not sure it looks better though (or if it works)

```suggestion
    if (!(options.dsn?.isEmpty == false)) {
```

Reply:
yeah still feels not so nice, I'd expect something like https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/is-null-or-empty.html